src/Cryptol/TypeCheck/Error.hs

```diff
@@ -109,6 +111,10 @@ data Error    = KindMismatch (Maybe TypeSource) Kind Kind
               | RecursiveTypeDecls [Name]
                 -- ^ The type synonym declarations are recursive
 
+              | TooManyParams Name Type Int Int
+                -- ^ A binder implies more arguments than
+                --   its type signature
```

Comment:
Please leave a comment saying what these fields represent. In particular, it's not obvious which `Int` is the expected number of arguments, and which `Int` is the actual number of arguments.